src/lib/storage/proposal/space_maker.rb

```diff
@@ -211,71 +273,93 @@ def shrink_partition(partition, shrink_size)
           partition.size = partition.size - shrink_size.size
         end
 
-        # Use force to create space: delete partitions while there is no
-        # suitable space distribution
+        # Use force to create space: delete partitions if a given type while
+        # there is no suitable space distribution
+        #
+        # @see #deletion_candidate_partitions for supported types
         #
-        # @param devicegraph [DeviceGraph] devicegraph to update
         # @param volumes [PlannedVolumesList]
+        # @param type [Symbol] type of partition to delete
         # @param keep [Array<String>] device names of partitions that should not
         #       be deleted
-        def delete_partitions!(devicegraph, volumes, keep)
+        # @param disk [String] optional disk name to restrict operations to
+        def delete_partitions!(volumes, type, keep, disk)
```

Comment:
Would it make sense to have an early 'if success?' here? If only to avoid repeatedly building of the partition list and not filling the log unnecessarily?